cpp/src/community/spectral_clustering.cu

```diff
@@ -21,24 +21,20 @@
  * ---------------------------------------------------------------------------**/
 
 #include <algorithms.hpp>
-#include <graph.hpp>
 
-#include <nvgraph/include/sm_utils.h>
 #include <rmm/thrust_rmm_allocator.h>
 #include <thrust/transform.h>
 #include <ctime>
-#include <nvgraph/include/nvgraph_error.hxx>
-#include <utilities/error.hpp>
 
-#include <nvgraph/include/modularity_maximization.hxx>
```

Comment:
I would suggest removing:
 * nvgraph/*.cu
 * nvgraph/*.cpp
Removing the corresponding entries from CMakeLists.txt.  If we're picking this all up from raft now, we shouldn't need these files.

I would suggest removing anything in nvgraph/include/* that is not referenced once you remove the above files.

That will make it clear what remaining nvgraph references we have internally.

Reply:
Done. `nvgraph` dependencies were removed (specifically, form: spectral_clustering, triangle_counting, extract_by_vertex, and louvain).

And `nvgraph` dir was moved under `src/legacy/`.